TestAssets/TestPackages/dotnet-dependency-tool-invoker/Program.cs

```diff
@@ -0,0 +1,83 @@
+﻿using System;
+using System.IO;
+using System.Collections.Generic;
+using System.Linq;
+using Microsoft.DotNet.ProjectModel;
+using Microsoft.DotNet.Cli.Utils;
+
+namespace Microsoft.DotNet.Tools.DependencyInvoker
+{
+    public class Program
+    {
+        public static void Main(string[] args)
+        {
+            DebugHelper.HandleDebugSwitch(ref args);
+
+            var dotnetParams = new DotnetBaseParams("dotnet-dependency-tool-invoker", "DotNet Dependency Tool Invoker", "Invokes tools declared as NuGet dependencies of a project");
+            
+            dotnetParams.Parse(args);
+            
+            if (string.IsNullOrEmpty(dotnetParams.Command))
+            {
+                Console.WriteLine("A command name must be provided");
+                
+                return;
+            }
+            
+            var projectContexts = 
+                CreateProjectContexts(dotnetParams.ProjectPath)
+                    .Where(p => dotnetParams.Framework == null ||
+                                dotnetParams.Framework.GetShortFolderName()
+                                .Equals(p.TargetFramework.GetShortFolderName()));
+            
+            var commandFactory =
+                new ProjectDependenciesCommandFactory(
+                    dotnetParams.Framework,
+                    dotnetParams.Config,
+                    dotnetParams.Output,
+                    dotnetParams.BuildBasePath,
+                    projectContexts.First().ProjectDirectory);
+                    
+            foreach (var projectContext in projectContexts)
+            {
+                Console.WriteLine($"Invoking '{dotnetParams.Command}' for '{projectContext.TargetFramework}'.");
+
+                try
+                {
+                    var exitCode = commandFactory.Create(
+                            $"dotnet-{dotnetParams.Command}",
+                            Enumerable.Empty<string>(),
+                            projectContext.TargetFramework,
+                            dotnetParams.Config)
+                        .ForwardStdErr()
+                        .ForwardStdOut()
+                        .Execute()
+                        .ExitCode;
+                    
+                    Console.WriteLine($"Command returned {exitCode}");
```

Comment:
The action to take is really up to the implementer. dotnet-test would fail if any fail, but EF only needs to succeed for one of these so it might fail if all fail. For the test helper I only care about console output, so leaving as is.